src/relayclient/GSNConfigurator.ts

```diff
@@ -71,17 +59,18 @@ function mergeDeep (target: any, ...sources: any[]): Object {
     return target
   }
   const source = sources.shift()
-
-  if (isObject(target) && isObject(source)) {
-    for (const key in source) {
-      if (Object.prototype.hasOwnProperty.call(source, key)) {
-        if (isObject(source[key])) {
-          if (target[key] == null) {
-            Object.assign(target, { [key]: {} })
+  if (source != null) {
```

Comment:
yuck. don't recurse - iterate.

Reply:
also - I didn't see a single test of this method.

Reply:
one last thing: you MODIFY your target parameter. its bad API (I know, shared with Object.assign, but still bad)
also, you don't use that fact - you pass in `{}` - so better remove it altogether, and just merge inputs without modifying any of them.

Reply:
or better: replace it with `const deepmerge = require('deepmerge')`

Reply:
This is a copy-pasted code. It is not strictly needed anymore, but I decided to keep it around in case there is any hierarchy in the GSNConfig again.

Reply:
so if you use it, then add a test.

Reply:
No

Reply:
ok. so remove it. unused, untested (and replacable by single standard method) do not belong to the code.

Reply:
Done